### PR TITLE
6 ➡️ 7 (main)

### PR DESCRIPTION
**Pull request by mjcarroll**
# ➡️ Forward port

Port ign-fuel-tools6 to main

Branch comparison: https://github.com/ignitionrobotics/ign-fuel-tools/compare/main...ign-fuel-tools6

**Note to maintainers**: Remember to **Merge** with commit (not squash-merge or rebase)

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -9,13 +9,13 @@ project(ignition-fuel_tools7 VERSION 7.0.0)
...
 # If you get an error at this line, you need to install ignition-cmake
-find_package(ignition-cmake2 2.3 REQUIRED)
+find_package(ignition-cmake2 2.8.0 REQUIRED)
 set(IGN_CMAKE_VER ${ignition-cmake2_VERSION_MAJOR})
 
 #============================================================================
 # Configure the project
 #============================================================================
-ign_configure_project(VERSION_SUFFIX pre1)
+ign_configure_project(VERSION_SUFFIX)
```
Comment (chapulina) [inline CMakeLists.txt:18]: We shouldn't remove the `pre1` from `main` yet

File: src/WorldIter.cc [hunk 2]
```diff
@@ -116,9 +117,12 @@ WorldIterRestIds::WorldIterRestIds(const Rest &_rest,
   std::string queryStrPageKey = "page=";
   do
   {
+    // Fire the request.
     // Fire the request.
```
Comment (adlarkin) [inline src/WorldIter.cc:121]: nit: remove duplicate comment?

File: src/ign.hh [hunk 3]
```diff
@@ -125,4 +127,15 @@ extern "C" IGNITION_FUEL_TOOLS_VISIBLE int editUrl(
     const char *_private = nullptr,
     const char *_path = nullptr);
 
+/// \brief External hook to execute 'ign fuel update [options] from the command'
+/// line
+///
+/// \param[in] _onlyModels "1" to only update models.
```
Comment (adlarkin) [inline src/ign.hh:133]: nit: remove empty line?

**Review by adlarkin — COMMENTED** [review]
LGTM, but Windows CI is still failing. Looking at the failure though, it shows that this is not a new failure (age of 21): https://build.osrfoundation.org/job/ign_fuel-tools-pr-win/61/testReport/

So, should Windows CI no longer be a required check? Or do we need to fix this test?

**Comment by adlarkin** [comment]
Looks like windows CI is still failing with https://github.com/ignitionrobotics/ign-fuel-tools/pull/201/commits/44c4b23a5def0bfd8ad4fddeb2d346c719c3d1eb:
```
C:\Jenkins\workspace\ign_fuel-tools-pr-win\ws\ign-fuel-tools\src\Helpers_TEST.cc:43
Expected equality of these values:
  testStr1
    Which is: "localhost:8000\\\\some\\\\path"
  uriToPath(uri)
    Which is: "localhost:8000\\some\\path"
```

**Comment by mjcarroll** [comment]
@adlarkin Windows test should be resolved, but I'm not sure if it's the "correct" answer here. 

While I understand that Windows uses backslashes for path separators, I don't believe that URIs should have to, in general.  Looking at [file URI scheme Wikipedia](https://en.wikipedia.org/wiki/File_URI_scheme) everything is using forward slashes.

Maybe @nkoenig or @ahcorde has some more insight?

**Comment by chapulina** [comment]
> While I understand that Windows uses backslashes for path separators, I don't believe that URIs should have to, in general. 

URIs should always use `/`, and local paths use each system's separator. So the input to `uriToPath` is the same for all platforms, but the returned value is platform-specific.

Another think to keep in mind is that Windows doesn't accept `:` in paths, so on https://github.com/ignitionrobotics/ign-common/pull/197 `common::joinPaths` was updated to strip that and other characters. So the expected path on Windows should be `localhost8000` instead of `localhost:8000`.

**Comment by mjcarroll** [comment]
> Another thing to keep in mind is that Windows doesn't accept : in paths, so on ignitionrobotics/ign-common#197 common::joinPaths was updated to strip that and other characters. So the expected path on Windows should be localhost8000 instead of localhost:8000.

That isn't what is currently happening on Windows, though.  I just modified the test, and these are the values that make it pass:

 `uriToPath(common::URI uri{"http://localhost:8000/some/path"});` maps to `localhost:8000\some\path`
 `uriToPath(common::URI uri{"http://localhost:8000/some/path/"});` maps to `localhost:8000\some\path\`

**Comment by mjcarroll** [comment]
I think the `uriToPath` should only accept `file://` URIs.  Everything else should probably fail?

**Comment by chapulina** [comment]
> That isn't what is currently happening on Windows, though. 

Yeah I think we haven't fixed `uriToPath` yet

> I think the uriToPath should only accept file:// URIs. Everything else should probably fail?

It looks like it's being used to convert the Fuel Server URL:

https://github.com/ignitionrobotics/ign-fuel-tools/search?q=uritopath

File: src/Helpers_TEST.cc [hunk 4]
```diff
@@ -25,21 +25,27 @@ using namespace fuel_tools;
...
-    EXPECT_EQ(common::joinPaths("localhost:8000", "some", "path"),
-        uriToPath(uri));
+    EXPECT_EQ(testStr1, uriToPath(uri));
   }
 
   {
     common::URI uri{"http://localhost:8000/some/path/"};
-    EXPECT_EQ(common::separator(common::joinPaths("localhost:8000", "some",
-        "path")), uriToPath(uri));
+    EXPECT_EQ(testStr2, uriToPath(uri));
```
Comment (Blast545) [inline src/Helpers_TEST.cc:48]: I think If this test is changed, the tests below with `UriToPathHasAuthority` should change as well

**Comment by Blast545** [comment]
@chapulina Windows CI is green with the changed tests in `0033e68`, do you think this issue with `uriToPath` should be addressed before merging [ign-common#246](https://github.com/ignitionrobotics/ign-common/pull/246) or at another time?

:man_farmer: I think it's ok to remove the tests as long as we add a note there and don't lose track of it. (I can open a follow up issue)

**Comment by chapulina** [comment]
> Windows CI is green with the changed tests in 0033e68

I think we can merge the PR as is for now, but I think we should come back to fix the expectations later, because I think `localhost:8000\some\path\` is not a valid Windows path.

> do you think this issue with uriToPath should be addressed before merging ign-common#246 or at another time?

I think we don't need to block that for this

**Comment by Blast545** [comment]
> I think we can merge the PR as is for now
Sure, I'll add an issue to fix the expectation at another time. Thanks!

File: src/Helpers_TEST.cc [hunk 5]
```diff
@@ -25,21 +25,27 @@ using namespace fuel_tools;
 /////////////////////////////////////////////////
 TEST(HelpersTEST, UriToPathNoAuthority)
 {
+#ifdef WIN32
```
Comment (Blast545) [inline src/Helpers_TEST.cc:28]: Please add a reference to the `uriToPath` issue here before merging.
```suggestion
// TO-DO: Update this test after ign-fuel-tools#204 is addressed
#ifdef WIN32
```

**Review by chapulina — APPROVED** [review]

File: src/Helpers_TEST.cc (hunk 4 above)
Comment (mjcarroll) [reply src/Helpers_TEST.cc:49]: ee7cae2

**Comment by chapulina** [comment]
@mjcarroll , can we clean up the commit history and get this in?

**Comment by mjcarroll** [comment]
> @mjcarroll , can we clean up the commit history and get this in?

Done!

**Pull request merged**